src/test/java/Page_Objects/SongPreviewPage.java

```diff
@@ -28,4 +38,51 @@ public void goToPlayTheSong() {
         playButton.click();
     }
 
+    public void toggleDifficultyLevel() {
+        difficultyModeElement.click();
+    }
+
+    public void expectGameDifficultyLevelToBe(String level) {
+        String difficultyLevelAttribute = difficultyModeElement.getAttribute("data-test-value");
+
+        Assertions.assertEquals(level, difficultyLevelAttribute);
+    }
+
+    public void toggleGameMode() {
+        gameModeElement.click();
+    }
+
+    public void expectGameModeToBe(String modeName) {
+        String gameModeAttribute = gameModeElement.getAttribute("data-test-value");
+
+        Assertions.assertEquals(modeName, gameModeAttribute);
+    }
+
+    public WebElement playerNameInput(int playerNumber) {
+        return driver.findElement(By.cssSelector("[data-test='player-" + playerNumber + "-name']"));
+    }
+
+    public void enterPlayerName(int playerNumber, String playerName) {
+        playerNameInput(playerNumber).sendKeys(playerName);
+    }
+
+    public void expectEnteredPlayerNameToBePrefilledWith(int playerNumber, String playerName) {
+        String playerNameAttribute = playerNameInput(playerNumber).getAttribute("placeholder");
+
+        Assertions.assertEquals(playerName, playerNameAttribute);
+    }
+
+    public boolean isTheNameOnTheRecentPlayerList(String expectedName) {
+        playerNameInput(0).click();

```

Comment:
this `0` should be an argument of the method

Reply:
I was thinking about the same, but decided to make it more 'simple', because the names in recent-player-list are the same for both players